mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h

```diff
@@ -1245,6 +1249,1321 @@ struct Conv2DRewritePattern : public OpRewritePattern<T> {
+                                                 strideH / gcdStrideDilationH),
+                                             b.getI32IntegerAttr(1),
+                                             b.getI32IntegerAttr(0),
+                                         })),
+            b.getNamedAttr(
+                "source_dimensions",
+                b.getArrayAttr({b.getI32IntegerAttr(filterKeyToDim["y"])})),
+            b.getNamedAttr("source_names",
+                           b.getArrayAttr({b.getStringAttr("y")}))};
+
+        // x
+        curOutputDimName.push_back(b.getStringAttr("xdot"));
+        curOutputDimName.push_back(b.getStringAttr("xtilda"));
+        transformedFilterShape.push_back(xDot);
+        transformedFilterShape.push_back(xTilda);
+        llvm::SmallVector<NamedAttribute, 6> xDimAttr{
+            b.getNamedAttr("dimensions",
+                           b.getArrayAttr({b.getI32IntegerAttr(5),
+                                           b.getI32IntegerAttr(6)})),
+            b.getNamedAttr("names", b.getArrayAttr({curOutputDimName[5],
+                                                    curOutputDimName[6]})),
+            b.getNamedAttr("transformation", b.getStringAttr("Embed")),
+            b.getNamedAttr("parameters", b.getArrayAttr({
+                                             b.getI32IntegerAttr(
+                                                 strideW / gcdStrideDilationW),
+                                             b.getI32IntegerAttr(1),
+                                             b.getI32IntegerAttr(0),
+                                         })),
+            b.getNamedAttr(
+                "source_dimensions",
+                b.getArrayAttr({b.getI32IntegerAttr(filterKeyToDim["x"])})),
+            b.getNamedAttr("source_names",
+                           b.getArrayAttr({b.getStringAttr("x")}))};
+
+        transformedFilterAttrs.push_back(b.getNamedAttr(
+            "layout",
+            b.getArrayAttr(
+                {b.getDictionaryAttr(gDimAttr), b.getDictionaryAttr(kDimAttr),
+                 b.getDictionaryAttr(cDimAttr), b.getDictionaryAttr(yDimAttr),
+                 b.getDictionaryAttr(xDimAttr)})));
+        transformedFilterAttrs.push_back(b.getNamedAttr(
+            "output_layout",
+            b.getArrayAttr(ArrayRef<Attribute>(curOutputDimName.begin(),
+                                               curOutputDimName.end()))));
+
+        transformedFilterAttrs.push_back(
+            b.getNamedAttr("source_layout", filterLayoutAttr));
+
+        auto transformedFilterMemRefType =
+            MemRefType::get(transformedFilterShape, filterElementType);
+        auto gemm =
+            b.create<miopen::TransformOp>(loc, transformedFilterMemRefType,
+                                          op.filter(), transformedFilterAttrs);
+        return gemm;
+      };
+
+      auto wei_g_k_c_ydot_ytilda_xdot_xtilda =
```

Comment:
`weightGKCYdotYtildaXdotXtilda` to conform with variable naming convention. the name of the lambda should also be changed.